src/xform_to_datalog/manifest_datalog_facts.h

```diff
@@ -24,16 +24,16 @@
 #include "src/ir/tag_check.h"
 #include "src/ir/tag_claim.h"
 #include "src/ir/datalog_print_context.h"
-#include "src/ir/particle_spec.h"
+#include "src/ir/system_spec.h"
 #include "third_party/arcs/proto/manifest.pb.h"
 
 namespace raksha::xform_to_datalog {
 
 class ManifestDatalogFacts {
  public:
   static ManifestDatalogFacts CreateFromManifestProto(
-      const arcs::ManifestProto &manifest_proto,
-      ir::ParticleSpecRegistry &particle_spec_registry);
+      const ir::SystemSpec* system_spec,
```

Comment:
Why are we using a `const ir::SystemSpec *` instead of a `const ir::SystemSpec &`? The latter would ensure that we cannot be passed a `nullptr`.

Reply:
Good point. Will do.